### DISCUSSION TITLE
Design system w/ Storybook for UI components

### DISCUSSION THREAD

**Discussion by JorchRL** [deleteman/the-hive#10] [Ideas]
I've been playing around with Storybook and think that maybe we could start building a design system with it. (using Brad Frost's Atomic Design methodology https://atomicdesign.bradfrost.com/chapter-2/)

In particular, we could build something like this one: https://www.kickstartds.com/storybook/?path=/story/welcome--page

I see some advantages:

- The hive's own design system. Brand identity and and UI patterns for its particular needs!
- We can iterate fast over components without messing with the live website.
- A unique (visual) identity to the website (and control over the license of the component library).
- Easier to develop and test each component individually.
- We can even develop entire pages (it integrates nicely with Next's pages) like a blog template for example 

It could either be done on a separate repo, or a separate branch. To avoid polluting this repo. I could take care of that (i think :sweat_smile: ) if you are ok with that. (it could be structured like this: https://storybook.js.org/blog/structuring-your-storybook/)

I think maybe it could be done with

- Storybook for developing and testing the components in isolation
- A headless UI library (instead of material UI, which is hard to override)
- Tailwind (or emotion, styled components)

And we could integrate those gradually into the app.

Let me know what you think :sweat_smile:

**Comment by deleteman** [comment]
Hey @JorchRL , I think the idea is great, thanks for the suggestion and the offer of doing it yourself!
We'll need this once the project is live and moving forward I agree.
Personally, if you want to move ahead and take this one on a separate project, by all means, do it and share the link, I'll contribute to it as much as I can, but my focus right now needs to be getting this thing working and published.

As you can see I don't have a lot of time to work on it (sadly), so I need to optimize the time I have, thus my priorities right now are:

1. Get it working with whatever styles I can get my hands on.
2. Get feedback from actual users
3. Worry about improving the UI.

Don't get me wrong, if I had a bigger team, UI / UX would be prior 1 or 2, but right now I don't have anything. So, again, if you're up for it, I'd love to see what you can do with storybook and we'll start integrating it into the main project once it's live and working.

**Reply by JorchRL** [reply]
Awesome! I tough about it so you can keep focusing on the main app, using the UI components already there. 

While I may be able to contribute something useful without taking your focus away 😅

I’ll get to it. Thanks
